### DISCUSSION TITLE
Keep dependencies up to date in all Octoherd scripts

### DISCUSSION THREAD

**Discussion by oscard0m** [octoherd/octoherd#34] [Ideas]
I would like to find a way to invite users to have dependencies up to date but not to enforce the usage of Renovate/Dependabot.

What do you think about using Octoherd's GitHub App to keep track and update dependencies of the project?

P.S.: I don't remember if the code of the GitHub App is public but I would be open to take a look into it and draft a proposal.

**Comment by gr2m** [comment]
> I would like to find a way to invite users to have dependencies up to date but not to enforce the usage of Renovate/Dependabot.

I fear that would add a huge amount of complexity. I'd rather encourage folks to keep their dependencies up-to-date.

At some point I'd love to have an octoherd website where folks can search for plugins, and on that website we could add indicators which dependencies a script is using and whether these dependencies are up-to-date. That might be a good incentive by the author to keep them updated.

Another thing is that we don't really have a good story for testing octoherd script. And until we make it simple to create / maintain tests, it will be hard to create confidence in merging automated pull requests for dependency updates.

> I don't remember if the code of the GitHub App is public

There is no code :) It's just a registered GitHub app that we could use in future for things like sending pull request to upgrade octoherd scripts
